### DISCUSSION TITLE
Audio-over-IP Support?

### DISCUSSION THREAD

**Discussion by 0lek** [micro-nova/AmpliPi#288] [Ideas]
Hey,

Even though I have managed to connect only 1 set of speakers so far, I already know I will probably want to get an extension. However, it will not cover all the use cases that I need. That's why I would like the AmpliPi to output to "satellite zones" - probably via Audio over IP or sth else.

In my specific use case, I will need to connect speakers on the terrace, in the office and in the garden shed. They do not have wiring (due to different reasons) from the rack. So my ideal situation would be to have a raspi with an audiohat (e.g. hifiberry or sth similar) and connect speakers directly to it. I want the ability to control this "zone" from the same interface as the directly connected zones. This would be a huge game-changer!

**Comment by linknum23** [comment]
Adding wireless zones is something we have considered but due to lack of hardware support for analog inputs it adds a lot of complexity on the software side. Realistically we would probably support a setup like this with a more advanced version of AmpliPi hardware.

Support for this would likely use an audio streaming protocol like Snapcast, check out the discussion here: https://github.com/micro-nova/AmpliPi/discussions/75

Highlights for a supported wireless setup (assuming you have extra unused zones):

> Have you considered using the new preamp outputs to a wireless transmitter? This could be advanced as a purpose-built rf transmitter + speaker pair, or be a wifi-based solution with a raspberry pi running snapcast server with a ADC (connected to the preamp) ---> a raspberry pi with a DAC + AMP connected to the outdoor speakers.

Or for supporting long wire-able distances:

> Another option you could consider is using an audio balun for transmission. This could be used to convert the preamp outputs to a signal that can be transmitted over longer distances using cat5. This would need to be connected to outdoor powered speakers. Here's one on Amazon: https://www.amazon.com/dp/B08JQHT14.

**Reply by linknum23** [reply]
I migrated this to a discussion, so we could keep talking about it from a "big idea" perspective.

**Reply by Lohrer** [reply]
Just to add to this, if some kind of audio pipes are added (see #262) then one of the audio sources could be split to play on both the AmpliPi's DAC and also to some IP streaming output. The audio pipe feature is far out on our roadmap though unless someone in the community wanted to pick it up.

**Comment by 0lek** [comment]
Hey,

The pre-amp outs could be a solution, I agree. I also don't have an issue with building a ADC/DAC pair on raspberrys. However, I won't be too happy to have another application to manage. I would love to have this transparently handled by AmpliPi, so I don't need to start amplipi app, then the snapcast app to play music out on the terrace. Is this something that will be available? I understand this is connected with #75 
Otherwise the WAF will go down significantly :)

**Comment by jonathanslenders** [comment]
Right now out of stock, and not (yet) tested myself, but these look nice and low latency too: https://www.audiopro.com/en/product/wf100tx-rx/
They take power from USB, but input can also come from analog stereo I believe. See here: https://www.trustedreviews.com/reviews/audio-pro-wf100-wireless-transmitter
(I'm considering this for playing from a laptop to the Amplipi.)
